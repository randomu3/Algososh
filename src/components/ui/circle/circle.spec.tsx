// link.test.js
import { Circle } from "./circle";
import { render, fireEvent, waitFor, screen } from "@testing-library/react";
import { ElementStates } from "../../../types/element-states";

describe("Проверяем при помощи снэпшотов корректную отрисовку элемента", () => {
  it("-> без буквы:", () => {
    render(<Circle letter={""} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });
  it("-> с буквами (Snapshot):", () => {
    render(<Circle letter={"test-text"} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> с head (Snapshot):", () => {
    render(<Circle head={"0"} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> с react-элементом в head:", () => {
    render(<Circle head={<>text</>} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> с tail:", () => {
    render(<Circle tail={"0"} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> с react-элементом в tail:", () => {
    render(<Circle tail={<>text</>} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> с index:", () => {
    render(<Circle index={0} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> с пропом isSmall === true:", () => {
    render(<Circle isSmall={true} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> в состоянии default:", () => {
    render(<Circle state={ElementStates.Default} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> в состоянии changing:", () => {
    render(<Circle state={ElementStates.Changing} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });

  it("-> в состоянии modified:", () => {
    render(<Circle state={ElementStates.Modified} />);
    const circle = screen.getByTestId("circle");
    expect(circle).toMatchSnapshot();
  });
});
