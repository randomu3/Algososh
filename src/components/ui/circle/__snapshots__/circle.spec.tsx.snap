// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> без буквы: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> в состоянии changing: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> в состоянии default: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> в состоянии modified: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с head (Snapshot): 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    0
  </div>
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с index: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-cy="circle-0"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с react-элементом в head: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    text
  </div>
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с react-элементом в tail: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    text
  </div>
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с tail: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    0
  </div>
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с буквами (Snapshot): 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    >
      test-text
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Проверяем при помощи снэпшотов корректную отрисовку элемента -> с пропом isSmall === true: 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
    data-cy="circle-undefined"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
